src/asset-swapper/quote_consumers/feature_rules/transform_erc20_rule.ts

```diff
@@ -99,74 +99,82 @@ export class TransformERC20Rule extends AbstractFeatureRule {
-                    data: encodeAffiliateFeeTransformerData({
-                        fees: [
-                            {
-                                token: isToETH ? ETH_TOKEN_ADDRESS : buyToken,
-                                amount: buyTokenFeeAmount,
-                                recipient: feeRecipient,
-                            },
-                        ],
-                    }),
-                });
-                // Adjust the minimum buy amount by the fee.
-                minBuyAmount = BigNumber.max(0, minBuyAmount.minus(buyTokenFeeAmount));
-            }
-            if (sellTokenFeeAmount.isGreaterThan(0)) {
-                throw new Error('Affiliate fees denominated in sell token are not yet supported');
-            }
-        } else if (feeType === AffiliateFeeType.GaslessFee && feeRecipient !== NULL_ADDRESS) {
-            if (buyTokenFeeAmount.isGreaterThan(0)) {
+        affiliateFees.forEach((affiliateFee) => {
+            const { feeType, buyTokenFeeAmount, sellTokenFeeAmount, recipient: feeRecipient } = affiliateFee;
+            if (feeType === AffiliateFeeType.PositiveSlippageFee && feeRecipient !== NULL_ADDRESS) {
+                // bestCaseAmountWithSurplus is used to cover gas cost of sending positive slipapge fee to fee recipient
+                // this helps avoid sending dust amounts which are not worth the gas cost to transfer
+                let bestCaseAmountWithSurplus = quote.bestCaseQuoteInfo.makerAmount
+                    .plus(
+                        POSITIVE_SLIPPAGE_FEE_TRANSFORMER_GAS.multipliedBy(quote.gasPrice).multipliedBy(
+                            quote.makerAmountPerEth,
+                        ),
+                    )
+                    .integerValue();
+                // In the event makerAmountPerEth is unknown, we only allow for positive slippage which is greater than
+                // the best case amount
+                bestCaseAmountWithSurplus = BigNumber.max(
+                    bestCaseAmountWithSurplus,
+                    quote.bestCaseQuoteInfo.makerAmount,
+                );
                 transformations.push({
-                    deploymentNonce: this.transformerNonces.affiliateFeeTransformer,
-                    data: encodeAffiliateFeeTransformerData({
-                        fees: [
-                            {
-                                token: isToETH ? ETH_TOKEN_ADDRESS : buyToken,
-                                amount: buyTokenFeeAmount,
-                                recipient: feeRecipient,
-                            },
-                        ],
+                    deploymentNonce: this.transformerNonces.positiveSlippageFeeTransformer,
+                    data: encodePositiveSlippageFeeTransformerData({
+                        token: isToETH ? ETH_TOKEN_ADDRESS : buyToken,
+                        bestCaseAmount: BigNumber.max(bestCaseAmountWithSurplus, quote.bestCaseQuoteInfo.makerAmount),
```

Comment:
```suggestion
                        bestCaseAmount: bestCaseAmountWithSurplus,
```
Since it already took the [max](https://github.com/0xProject/0x-api/pull/1225/files#diff-29142bc7988bad3cc0b321c1151ff7e6e3017fcce9d200b0fbf0fcbd78a1f219R117)